src/ui/popup/components/site-toggle/index.tsx

```diff
@@ -25,10 +27,10 @@ export default function SiteToggleButton({data, tab, actions}: ExtWrapper & {tab
         <Button
             class={{
                 'site-toggle': true,
-                'site-toggle--active': isSiteEnabled,
+                'site-toggle--active': isSiteEnabled || PDF,
                 'site-toggle--disabled': !toggleHasEffect
             }}
-            onclick={() => actions.toggleURL(tab.url)}
+            onclick={() => PDF ? actions.changeSettings({enableForPDF: !data.settings.enableForPDF}) : actions.toggleURL(tab.url)}
```

Comment:
Good, but I feel like it's better to move such long lines out of attribute. Declare an event handler somewhere above.

The problem with JSX is that it becomes less readable with complex logic inside. Also when the layout changes, all this logics shifts too. This makes it harder to use tools like `git blame`, or review PRs.